### PR TITLE
Laravel 5 rewrite and test addition

### PR DESCRIPTION
**Pull request by namelivia**
I got this ready for fixing #4 , it comes with a lot of chages so it should be carefully checked.
This is a summary of what I've done:
- StyleCI added for code style checking and adapted the code for it.
- Unit and integration tests for almost everything, executed through phpunit and TravisCI.
- The code has been reorganized for better testeability and (hopefully) readability and extendability. This means for example that adding new api methods should easy by adding short functions for them to `FatSecret.php`.
- API error handling by throwing custom Exceptions
- Complete integration test by calling all the methods from an example laravel 5 app to check they work.
Feel free to comment and criticize everything you don't like

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 FatSecret API for Laravel
 ============================
+[![StyleCI](https://styleci.io/repos/107026275/shield?branch=laravel5)](https://styleci.io/repos/107026275)
+[![Build Status](https://travis-ci.org/namelivia/fatsecret-laravel.svg?branch=laravel5)](https://travis-ci.org/namelivia/fatsecret-laravel)
```
Comment (Braunson) [inline README.md]: These badges are not for this repository

File: README.md [hunk 2]
```diff
@@ -22,39 +24,38 @@ How to Install
...
+2. Go to recently created `config/fatsecret.php` file and add this in with your details in the provided array
 
 	```php
 	// API Key & Secret (http://platform.fatsecret.com)
-    'fatsecret' => [
-        'secret' => env('FATSECRET_SECRET'),
-        'key' => env('FATSECRET_KEY'),
+    return [
+			'key' => 'YOUR-API-KEY',
+			'secret' => 'YOUR-API-SECRET'
```
Comment (Braunson) [inline README.md]: These should support being set via environmental variables in the `.env` file.

**Review by Braunson — CHANGES_REQUESTED** [review]

**Comment by Braunson** [comment]
@namelivia Is this still available, I totally forgot about this PR and wanted to test it to have it merged as V2!

**Comment by Braunson** [comment]
@namelivia Following up once more, are you able to add this PR back, I'd like to merge it in after a quick test?

**Comment by namelivia** [comment]
This PR can't be reopened. However, next week I'll have some spare time so I can take a look at it and come up with something.

**Comment by namelivia** [comment]
I recreated the changes on #10 , I've checked that the unit tests are passing but haven't fully tested it with the actual API (I did that when I opened the last PR so I assume it will still work).

**Comment by Braunson** [comment]
@namelivia Thank you so much. I just made some minor adjustments, going to test with live credentials now on 5.2 then will make a release for 5.x and then another for 6.x.

I'm curious what version of Laravel are you running this package on?

**Comment by namelivia** [comment]
There are some versioning issues, some Laravel versions require some phpunit, mockery and php versions that will require some minor changes on each versions of the code. That's why the build is failing, I'm currently setting those up, we can continue the discussion in the currently open PR.

**Pull request closed without merging**